File: code/NorthJutlandBoundary/NorthJutlandBoundary.R
# Read in level-1 shapefile obtained from https://gadm.org/download_country_v3.html
sf <- readRDS('gadm36_DNK_1_sf.rds')

# Confirm entry 3 is 'North Jutland' region
sf[3,'VARNAME_1']

# Confirm there are 28 polygons for this region
length(sf[3,'geometry'][[1]])

# Write each polygon out to a different csv file
for(i in 1:28) {
    write.csv(sf[3,'geometry'][[1]][[i]],paste0('NorthJutlandPolygon',i,'.csv'))
}

File: code/NorthJutlandBoundary/NorthJutlandBoundary.py
#!/usr/bin/env python3
import pickle
import numpy as np
import matplotlib.pyplot as plt
from shapely.geometry import Point 
from shapely.geometry.polygon import Polygon

# Read in the polygons for 'North Jutland' from the csv files
polygons = []
for i in range(1,29):
    dat = np.genfromtxt(f'NorthJutlandPolygon{i}.csv', delimiter=',',skip_header=True)
    x = dat[:,1]
    y = dat[:,2]
    polygons.append(Polygon(zip(x,y)))

# Check we've done the right thing
for p in polygons:
    x,y = zip(*p.boundary.coords)
    plt.plot(x,y)

plt.show()


# Save polygons for 'North Jutland' to file
open('NorthJutlandBoundary.pickle','wb').write(pickle.dumps(polygons))
